### PR TITLE
[5.2] ArchiveModel: Deprecate and fix getData()

### PR DESCRIPTION
**Pull request by Hackwar**
### Summary of Changes
The ArchiveModel class of com_content contains a method getData(), which seems not have been updated in a long time and which also isn't used anywhere. This PR deprecates the method and replaces the content with a call to getItems().

### Testing Instructions
Codereview

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [X] Pull Request link for manual.joomla.org: <link> **MISSING**
- [] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by HLeithner** [comment]
@brianteeman what's the issue?

**Comment by brianteeman** [comment]
Missing documentation

**Comment by Hackwar** [comment]
What documentation are you expecting?

File: components/com_content/src/Model/ArchiveModel.php [hunk 1]
```diff
@@ -135,26 +135,12 @@ protected function getListQuery()
...
-            $limit      = $app->getInput()->get('limit', $params->get('display_num', 20), 'uint');
-            $limitstart = $app->getInput()->get('limitstart', 0, 'uint');
-
-            $query = $this->_buildQuery();
-
-            $this->_data = $this->_getList($query, $limitstart, $limit);
-        }
-
-        return $this->_data;
+        return $this->getItems();
```
Comment (HLeithner) [inline components/com_content/src/Model/ArchiveModel.php:146]: please trigger_error user deprecation warning

**Comment by pe7er** [comment]
Pull Request link for manual.joomla.org: https://github.com/joomla/Manual/pull/267

File: components/com_content/src/Model/ArchiveModel.php (hunk 1 above)
Comment (Hackwar) [reply components/com_content/src/Model/ArchiveModel.php:146]: Done

**Comment by pe7er** [comment]
Looks ok from code review.

**Comment by laoneo** [comment]
Thanks!

**Pull request merged**